src/core/Akka/Actor/ActorRefProvider.cs

```diff
@@ -225,34 +225,26 @@ public IActorRef RootGuardianAt(Address address)
 
         private LocalActorRef CreateUserGuardian(LocalActorRef rootGuardian, string name)   //Corresponds to Akka's: override lazy val guardian: LocalActorRef
         {
-            return CreateRootGuardianChild(rootGuardian, name, () =>
-            {
-                var props = Props.Create<GuardianActor>(UserGuardianSupervisorStrategy);
+            var cell = rootGuardian.Cell;
+            cell.ReserveChild(name);
+            var props = Props.Create<GuardianActor>(UserGuardianSupervisorStrategy);
```

Comment:
The previous method we used here, which was a bit of a YAGNI thing to begin with, caused the mono runtime to randomly fail on some occasions due to an expression compiler bug.